pandas/core/config_init.py

```diff
@@ -957,3 +960,24 @@ def register_converter_cb(key):
         styler_environment,
         validator=is_instance_factory([type(None), str]),
     )
+
+
+merge_chunk_size_doc = """
+: int (> 0)
+    Conditional merge chunked cross merge with deferred filter, chunk size
+"""
+
+
+def int_greater_than_0(x: Any):
+    if not (isinstance(x, int) and x > 0):
+        raise ValueError("option must be an int greater than 0")
+
+
+with cf.config_prefix("conditional_merge"):
+    cf.register_option(
+        "left_chunk_size", 1000, merge_chunk_size_doc, int_greater_than_0
```

Comment:
Shouldn't the chunk sizes be equal when performing the merge?

Reply:
they don't necessarily need to be (not sure of any practical reasons not to be so though), however, I've updated to just a single `chunk_size` option, can either be an `int` in which case left/right will be both be equal to that int, or can be a tuple[int, int] for left/right chunk size